cmake/match.cmake

```diff
@@ -6,32 +6,64 @@
 # with a known good match file.
 #
 
-if(NOT DEFINED TEST_FILE)
-    message(FATAL_ERROR "TEST_FILE needs to be defined")
-endif()
-if(NOT DEFINED MATCH_FILE)
-    message(FATAL_ERROR "MATCH_FILE needs to be defined")
+# Process variables passed to the script
+if(NOT DEFINED MODE)
+    message(FATAL_ERROR "MODE needs to be defined. Possible values: 'binary', 'file', and 'nofile'")
+elseif(${MODE} STREQUAL "binary")
+    if(NOT DEFINED TEST_FILE)
+        message(FATAL_ERROR "TEST_FILE needs to be defined with a path to a binary to run")
+    else()
+        set(OUT_FILE "_matchtmpfile")
+    endif()
+elseif(${MODE} STREQUAL "file" OR ${MODE} STREQUAL "nofile")
+    if(NOT DEFINED OUT_FILE)
+        message(FATAL_ERROR "OUT_FILE needs to be defined with an output file to be verified")
+    endif()
+else()
+    message(FATAL_ERROR "${MODE} mode not recognised. Possible values: 'binary', 'file', and 'nofile'")
 endif()
 
-set(TEST_OUT "_matchtmpfile")
+if(NOT ${MODE} STREQUAL "nofile")
+    if(NOT DEFINED MATCH_FILE)
+        message(FATAL_ERROR "MATCH_FILE needs to be defined")
+    endif()
+endif()
 
-execute_process(
-    COMMAND ${TEST_FILE}
-    OUTPUT_FILE ${TEST_OUT}
-    RESULT_VARIABLE TEST_RESULT
-)
+# In 'nofile' mode, check if the OUT_FILE exists and exit the script
+if(${MODE} STREQUAL "nofile")
+    if(EXISTS ${OUT_FILE})
+        file(REMOVE ${OUT_FILE})
```

Comment:
you souldn't remove random files like this.

Reply:
Moved removal of files to tests set up. However, the output file is not removed after tests, rather before them, so the out file will exist after running the last test.